pkg/tlscheck/tlscheck.go

```diff
@@ -18,9 +19,24 @@ const (
 	timeout = 2 * time.Second
 )
 
+// addrValid validates the URL.
+// It returns an error if addr contains scheme prefix.
+func addrValid(addr string) error {
+	if strings.Contains(addr, "://") {
+		return fmt.Errorf("URL %q should not contain scheme prefix", addr)
+	}
```

Comment:
Since `TrimHTTPPrefixes` only trims `http` and `https`, could we have a problem if for whatever reason the address contains other prefix (`ftp://`, `tcp://` or something like that)? Maybe it's ok to just treat those as invalid.

Reply:
Yes, this error is intentional. If we keep `ftp://` prefix, then the `netutil.ParseEndpoint` would fail. I think it is better to fail early in that case.

Other idea: move the validation entirely to `netutil.ParseEndpoint`, but that has a risk of changing its behavior and this is used in many places.

Reply:
That's fair. I think it's ok to leave like this 👍

Reply:
Here is the behavior if we provide unsupported prefix: `kubernetes/listener/resources: 2024/06/28 08:59:25.052201 secrets.go:313: Error: unable to upsert registry "ftp://example2.com/" into store: unable to check TLS for registry "ftp://example2.com/": expected port name or number following ':' character`

(I think, this is not for the most current commit, but I will deploy the current one and check that)